src/utils/listenerHelpers.ts

```diff
@@ -0,0 +1,97 @@
+import { DispatchedEvent } from "globals/Calc";
+import { Calc } from "globals/window";
+
+const dispatchOverridingHandlers: [
+  (evt: DispatchedEvent) => boolean | undefined,
+  number,
+  number
+][] = [];
+
+let dispatchOverridingHandlerId = 0;
+
+// schedule a function to run after every desmos event
+// priorities determine which run first
+// the handler can return false to force the dispatcher to stop early
+// (e.g. to stop desmos from doing a default action upon pressing a key)
+export function registerCustomDispatchOverridingHandler(
+  handler: (evt: DispatchedEvent) => boolean | undefined,
+  priority: number
+): number {
+  const id = dispatchOverridingHandlerId++;
+  // add the handler
+  dispatchOverridingHandlers.push([handler, priority, id]);
+
+  // sort the handlers so that higher priorities are first
+  // could easily be optimized but prob not a bottleneck
+  dispatchOverridingHandlers.sort((a, b) => b[1] - a[1]);
+
+  return id;
+}
+
+// deregisters a function created with registerCustomDispatchOverridingHandler
+// uses the id that the former function returns
+export function deregisterCustomDispatchOverridingHandler(id: number): void {
+  // find the handler by id
+  const firstIndex = dispatchOverridingHandlers.findIndex(
+    ([_, __, id2]) => id2 === id
+  );
+  if (firstIndex === -1) {
+    // eslint-disable-next-line no-console
+    console.error(
+      `Failed to deregister the dispatch overriding handler '${id}'.`
+    );
```

Comment:
This is a reasonable place for console. Use (and import) `Console` instead of `console`.